torch/optim/optimizer.py

```diff
@@ -42,9 +65,14 @@ class Optimizer(object):
             options (used when a parameter group doesn't specify them).
     """
 
+    _optimizer_step_pre_hooks: Dict[int, Callable]
+    _optimizer_step_post_hooks: Dict[int, Callable]
+
     def __init__(self, params, defaults):
         torch._C._log_api_usage_once("python.optimizer")
         self.defaults = defaults
+        super().__setattr__('_optimizer_step_pre_hooks', OrderedDict())
```

Comment:
nit: no need to use super() here. You can just do `self._optimizer_step_pre_hooks = OrderedDict()`.